### ISSUE TITLE
ml_program support

### ISSUE THREAD

**Issue by silvasean** [openxla/stablehlo#182]
### Request description

[ml_program](https://discourse.llvm.org/t/rfc-introduce-ml-program-dialect-and-top-level-ops-proposal-v2/60907/72) is being used for certain JAX -> IREE flows and we are looking at using it in Torch-MLIR. Is there a plan for supporting this in StableHLO?

**Comment by joker-eph** [comment]
What does it mean to "support it"? What would be the difference with a `stablehlo.func` / `stablehlo.graph` operation?

**Comment by silvasean** [comment]
I mostly would be interested in the globals. In Torch-MLIR we still use func.func

**Comment by burmako** [comment]
Hi Sean! I'd also like to ask what kind of support you have in mind. 

[From the looks of it](https://github.com/llvm/llvm-project/blob/258d7b86eeab4222c783c1a19acd8e18cd7745aa/mlir/include/mlir/Dialect/MLProgram/IR/MLProgramOps.td#L180), `ml_program.global_load`, `ml_program.global_store` and friends work with the builtin `tensor` type, which is compatible with `stablehlo`. Although, I see that the ODS for these `ml_program` ops says `AnyType` which makes me worried a bit that I might've missed something.

**Comment by silvasean** [comment]
Yes, I think they are intended to be more general than just tensors. E.g. for storing a scalar `i64` for a counter / rng seed / etc.

**Comment by silvasean** [comment]
I would expect that we would close this PR when appropriate docs and specs have been updated to include ml_program in the set of "what is included in StableHLO"
